pkg/plugins/runtime/k8s/controllers/namespace_controller.go

```diff
@@ -27,17 +32,20 @@ type NamespaceReconciler struct {
 	Log logr.Logger
 
 	SystemNamespace     string
+	CNIEnabled          bool
 	ResourceManager     core_manager.ResourceManager
 	DefaultMeshTemplate mesh_proto.Mesh
 }
 
+// Reconcile is in charge for two things:
+//   - create NetworkAttachmentDefinition if CNI enabled and namespace has label 'kuma.io/sidecar-injection: enabled'
+//   - create 'default' mesh for the namespace 'SystemNamespace'
 func (r *NamespaceReconciler) Reconcile(req kube_ctrl.Request) (kube_ctrl.Result, error) {
-	if req.Name != r.SystemNamespace {
+	if !r.CNIEnabled && req.Name != r.SystemNamespace {
```

Comment:
It's a little bit difficult to follow logic of this method. Can we do a following
```
if req.Name == r.SystemNamespace {
  logic for creating a default mesh
}
if r.CNIEnabled {
  log for CNI. It should be ok if we execute this logic for SystemNamespace since it cannot have kuma.io/sidecar-injection anyways and if it does the Kuma will break anyways.
}
```

Reply:
I don't think we need this if anymore (on line 44)